libportability-gfx/src/impls.rs

```diff
@@ -1058,34 +1058,41 @@ pub extern "C" fn gfxQueueSubmit(
     pSubmits: *const VkSubmitInfo,
     fence: VkFence,
 ) -> VkResult {
-    assert_eq!(submitCount, 1); // TODO;
-
-    let submission = unsafe { *pSubmits };
-    let cmd_slice = unsafe {
-        slice::from_raw_parts(submission.pCommandBuffers, submission.commandBufferCount as _)
-    };
-    let wait_semaphores = unsafe {
-        let semaphores = slice::from_raw_parts(submission.pWaitSemaphores, submission.waitSemaphoreCount as _);
-        let stages = slice::from_raw_parts(submission.pWaitDstStageMask, submission.waitSemaphoreCount as _);
-
-        stages.into_iter()
-            .zip(semaphores)
-            .map(|(stage, semaphore)| (&**semaphore, conv::map_pipeline_stage_flags(*stage)))
-    };
-    let signal_semaphores = unsafe {
-        slice::from_raw_parts(submission.pSignalSemaphores, submission.signalSemaphoreCount as _)
-            .into_iter()
-            .map(|semaphore| &**semaphore)
+    let submits = unsafe {
+        slice::from_raw_parts(pSubmits, submitCount as usize)
     };
+    for (i, submission) in submits.iter().enumerate() {
+        let cmd_slice = unsafe {
+            slice::from_raw_parts(submission.pCommandBuffers, submission.commandBufferCount as _)
+        };
+        let wait_semaphores = unsafe {
+            let semaphores = slice::from_raw_parts(submission.pWaitSemaphores, submission.waitSemaphoreCount as _);
+            let stages = slice::from_raw_parts(submission.pWaitDstStageMask, submission.waitSemaphoreCount as _);
 
-    let submission = hal::queue::Submission {
-        command_buffers: cmd_slice.iter(),
-        wait_semaphores,
-        signal_semaphores,
-    };
+            stages.into_iter()
+                .zip(semaphores)
+                .map(|(stage, semaphore)| (&**semaphore, conv::map_pipeline_stage_flags(*stage)))
+        };
+        let signal_semaphores = unsafe {
+            slice::from_raw_parts(submission.pSignalSemaphores, submission.signalSemaphoreCount as _)
+                .into_iter()
+                .map(|semaphore| &**semaphore)
+        };
 
-    unsafe { queue.submit(submission, fence.as_ref()); }
+        let submission = hal::queue::Submission {
+            command_buffers: cmd_slice.iter(),
+            wait_semaphores,
+            signal_semaphores,
+        };
 
+        // only provide the fence for the last submission
+        let fence = if i + 1 == submits.len() {
+            fence.as_ref()
+        } else {
+            None
+        };
+        unsafe { queue.submit(submission, fence); }
```

Comment:
Could you add a TODO here?
In the future we probably don't want to everything one by one

Reply:
It is possible, but somehow I find this very low priority.